src/libraries/Microsoft.Extensions.Caching.Memory/src/MemoryCache.cs

```diff
@@ -500,5 +506,29 @@ private static void ValidateCacheKey(object key)
 
             static void Throw() => throw new ArgumentNullException(nameof(key));
         }
+
+        private class CoherentState
+        {
+            internal ConcurrentDictionary<object, CacheEntry> _entries = new ConcurrentDictionary<object, CacheEntry>();
+            internal long _cacheSize;
+
+            private ICollection<KeyValuePair<object, CacheEntry>> EntriesCollection => _entries;
```

Comment:
(nit) Is it worth it to go through an interface here? This property is only used in one spot, you could easily remove this and replace line 523 with `if (_entries.TryRemove(entry.Key, out _))`.

Reply:
iirc this lib targets .NET Standard 2.0 and using:

https://github.com/dotnet/runtime/blob/4e2d3a6f45f0808e42b5ac55b764e2e813e8a4c7/src/libraries/System.Collections.Concurrent/src/System/Collections/Concurrent/ConcurrentDictionary.cs#L1579

is the only way to remove the key from the dictionary if both key and value match.

The other method allows for removing key, no matter if values match or not:

https://github.com/dotnet/runtime/blob/4e2d3a6f45f0808e42b5ac55b764e2e813e8a4c7/src/libraries/System.Collections.Concurrent/src/System/Collections/Concurrent/ConcurrentDictionary.cs#L303

Reply:
> And the one that does not require an interface dispatch targets net5.0+:

Note, I'm actively working on #54012, which will add up-to-date TFMs to all extension libraries. So in the future we can use the best APIs available.